### PR TITLE
quota: calculating registry size (PROJQUAY-5476)

### PR DESCRIPTION
**Pull request by bcaton85**
Allows superusers to trigger a calculation of the deduplicated registry size. A superuser can go to the organization panel of the superuser page and select `Calculate` to queue a calculation of the registry total. The total will only be calculated when requested. Includes warning to user of increase of database load when running calculation.

### CODE REVIEW

File: data/database.py [hunk 1]
```diff
@@ -1012,6 +1012,13 @@ class QuotaRepositorySize(BaseModel):
     backfill_complete = BooleanField(null=False, default=False)
 
 
+class QuotaRegistrySize(BaseModel):
+    size_bytes = BigIntegerField(null=False, default=0)
+    running = BooleanField(null=False, default=False)
+    queued = BooleanField(null=False, default=False)
```
Comment (kleesc) [inline data/database.py:1018]: Why do you need a `queued` state if  there's already a `running` state? i.e what's the difference between use cases between the two.

**Review by kleesc — COMMENTED** [review]
.

File: data/database.py [hunk 2]
```diff
@@ -1012,6 +1012,13 @@ class QuotaRepositorySize(BaseModel):
...
+    completed_ms = BigIntegerField(null=True)
```
[the remark this answers is not in the document]
Comment (bcaton85) [reply data/database.py:1019]: the timestamp of when it was completed

File: data/database.py (hunk 1 above)
Comment (kleesc) [reply data/database.py:1018]: Couldn't you do that with only one state, with either `running` set or not?

Comment (bcaton85) [reply data/database.py:1018]: well if running is false it doesn't mean it should be ran. There needs to be a field saying whether the calculation is currently running or not so no other job picks it up and another field saying that the calculation is queued to be ran. If both are false nothing should happen. So really there's 3 states - do nothing, queued, running. I don't think we can represent that as a single boolean.

Comment (bcaton85) [reply data/database.py:1018]: this isn't a queue worker, it's a regular worker that periodically checks the table. This way when attempting to queue if it's already been queued it's just a single field check whether `queued == true`.

**Review by kleesc — APPROVED** [review]

**Pull request merged**